platforms/android/src/org/crosswalk/engine/XWalkWebViewEngine.java

```diff
@@ -161,6 +167,36 @@ public void onResume(boolean multitasking) {
         };
         pluginManager.addService(new PluginEntry("XWalkActivityDelegate", activityDelegatePlugin));
 
+        ArrayList<String> dangerous_permissions = new ArrayList<String>();
```

Comment:
It's better to hold those codes in a function,  and please confirm the feature of choosing file is work finely with App-tools.